### PR TITLE
Create .mergify.yml

### PR DESCRIPTION
**Pull request by azidar**
**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**
Adds mergify to facilitate backporting and merging PRs.

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      backport:
+        branches:
+          - 3.2.x
+      label:
+        add: [Backported]
+
+  - name: label Mergify backport PR
+    conditions:
+      - base=3.2.x
+      - body~=This is an automated backport of pull request \#\d+ done by Mergify.io
```
Comment (seldridge) [inline .mergify.yml]:
```suggestion
      - body~=This is an automated backport of pull request \#\d+ performed in the hall of the robot king
```

Comment (jackkoenig) [reply .mergify.yml]: 😂

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+pull_request_rules:
+  - name: remove outdated reviews
+    conditions:
+      - base=master
+    actions:
+      dismiss_reviews:
```
Comment (jackkoenig) [inline .mergify.yml]: I see we also have this on firrtl, but what does this rule do? A naive reading would be that it always just dismisses reviews on master but it clearly doesn't do that...

Comment (azidar) [reply .mergify.yml]: Looks like it enables mergify to take the last approved review; a stale review isn't considered if there is another approving one. https://doc.mergify.io/examples.html?highlight=dismiss#id11

**Pull request merged**